nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -159,34 +160,40 @@ BtNavigator::on_shutdown(const rclcpp_lifecycle::State & /*state*/)
 void
 BtNavigator::navigateToPose(const std::shared_ptr<GoalHandle> goal_handle)
 {
-  // Initialize the goal and result
-  auto goal = goal_handle->get_goal();
+  // The action may be pre-empted, so keep a pointer to the current goal handle
+  std::shared_ptr<GoalHandle> current_goal_handle = goal_handle;
   auto result = std::make_shared<nav2_msgs::action::NavigateToPose::Result>();
 
-  RCLCPP_INFO(get_logger(), "Begin navigating from current location to (%.2f, %.2f)",
-    goal->pose.pose.position.x, goal->pose.pose.position.y);
-
-  // Update the goal pose on the blackboard
-  *(blackboard_->get<geometry_msgs::msg::PoseStamped::SharedPtr>("goal")) = goal->pose;
+  initializeGoalPose(current_goal_handle);
 
   // Execute the BT that was previously created in the configure step
-  auto is_canceling = [goal_handle]() -> bool {return goal_handle->is_canceling();};
-  nav2_tasks::BtStatus rc = bt_->run(tree_, is_canceling);
+  auto is_canceling = [&current_goal_handle]() -> bool
+    {return current_goal_handle->is_canceling();};
+
+  auto on_loop = [this, &current_goal_handle, &result] {
+      if (action_server_->preempt_requested()) {
+        current_goal_handle->abort(result);
```

Comment:
@mjeronimo How do you think this should work? Since simple_action_server is managing the preemption policy, it kind of seems like it should also manage the abort of the old goal handle.

Reply:
@mjeronimo @crdelsey I pushed a change to manage the abort of the old goal to the server. 

Also, we omitted the abort in a few places, so this should fix it.

https://github.com/ros-planning/navigation2/blob/31c6f9393ed079daef56ed73713fbbdb9a2337a7/nav2_dwb_controller/dwb_controller/src/dwb_controller.cpp#L173

Reply:
@crdelsey Yes, I think it should manage the abort of the old goal. We have to correctly handle any accepted goals and dispense then appropriately. Deciding what the policy is is the tricky part.

Reply:
Let's track improvements to the action server in #811.